### DISCUSSION TITLE
strawberry shake: create client instance by constructor

### DISCUSSION THREAD

**Discussion by SvenBM** [ChilliCream/hotchocolate#4518] [Q&A]
Hi there,

how can i instantiate the generated client without dependency injection?
There's only 1 ctor that needs a bunch of parameters and i don't know how to get them properly wired up.
Can you provide an example (maybe update the documentation too) ?

btw: when i set "dependencyInjection": false the DI is still working?!

**Comment by repne** [comment]
@michaelstaib sorry for pinging you on this, I'm testing the new generate command on HC 13 and I can't find an answer to this question either. The `dependencyInjection` setting doesn't seem to make a difference in the generated code.

My requirement is pretty simple, I want to create many instances of the same client with different authentication tokens each time. I have a suite of service tests I'm porting to Strawberry Shake where the pattern followed is to create users on the fly for each test. Plus I have quite a few tests of different users interacting with each other (think of a websocket chat between users).

**Reply by repne** [reply]
After more research, this issue is related to #3871 and #4754. Those seems to be easy fixes, I'm happy to work on both if somebody can point me in the right direction.
Easiest way to implement this is to expose a client factory that accept the underlying http/websocket client (or that lets us configure it).

**Reply by michaelstaib** [reply]
Are you on slack?

https://slack.chillicream.com 

join and ping me there, the team coordinates there.

**Comment by kolpav** [comment]
@repne Have you made any progress on this? I was able to make SH to use my own http and websocket clients. Http one is somewhat easy but to do the same for websockets you need to implement your own IWebSocketClient which is the main paint point.
